packages/components/molecules/f-mega-modal/src/components/MegaModal.vue

```diff
@@ -43,13 +47,19 @@
                         <f-button
                             is-icon
                             :class="[$style['c-megaModal-closeBtn'], {
-                                [$style['c-megaModal-closeBtn--fixed']]: isCloseFixed || isFullHeight
+                                [$style['c-megaModal-closeBtn--fixed']]: isCloseFixed || isFullHeight,
+                                [$style['c-megaModal-backBtn']]: hasBackButton,
+                                [$style['c-megaModal-modeRTL']]: hasModeRightToLeft
```

Comment:
You should be able to remove these two lines once you move the classes onto the `c-megaModal` element as above.

Can just target:

```scss
.c-megaModal-modeRTL {
  & .c-megaModal-closeBtn { … }
}
```